sw/device/lib/runtime/epmp.h

```diff
@@ -0,0 +1,458 @@
+  kEpmpPermissionsUnlockedMachineAllUserExecute,     /**< LRWX = 0b0001 */
+  kEpmpPermissionsUnlockedMachineAllUserRead,        /**< LRWX = 0b0100 */
+  kEpmpPermissionsUnlockedMachineAllUserReadExecute, /**< LRWX = 0b0101 */
+  kEpmpPermissionsUnlockedMachineAllUserReadWrite,   /**< LRWX = 0b0110 */
+  kEpmpPermissionsUnlockedMachineAllUserAll,         /**< LRWX = 0b0111 */
+
+  // Locked permissions (can only be modified in machine mode when mseccfg.RLB
+  // is set).
+  kEpmpPermissionsLockedMachineNoneUserNone,               /**< LRWX = 0b1000 */
+  kEpmpPermissionsLockedMachineExecuteUserExecute,         /**< LRWX = 0b1001 */
+  kEpmpPermissionsLockedMachineReadUserRead,               /**< LRWX = 0b1100 */
+  kEpmpPermissionsLockedMachineReadExecuteUserReadExecute, /**< LRWX = 0b1101 */
+  kEpmpPermissionsLockedMachineReadWriteUserReadWrite,     /**< LRWX = 0b1110 */
+  kEpmpPermissionsLockedMachineAllUserAll,                 /**< LRWX = 0b1111 */
+} epmp_permissions_t;
+
+/**
+ * EPMP region specification.
+ *
+ * Provides the start and end addresses of a particular region. These addresses
+ * are byte-aligned (i.e. they are like regular pointers rather than encoded
+ * addresses).
+ *
+ * The intention is that this data structure is used to disambiguate the
+ * addresses of regions. It is therefore recommended that `start` and `end` are
+ * always used as labels when declaring a region.
+ */
+typedef struct epmp_region {
+  uintptr_t start;
+  uintptr_t end;
+} epmp_region_t;
+
+/**
+ * EPMP entry index.
+ */
+typedef enum epmp_entry {
+  kEpmpEntry0,
+  kEpmpEntry1,
+  kEpmpEntry2,
+  kEpmpEntry3,
+  kEpmpEntry4,
+  kEpmpEntry5,
+  kEpmpEntry6,
+  kEpmpEntry7,
+  kEpmpEntry8,
+  kEpmpEntry9,
+  kEpmpEntry10,
+  kEpmpEntry11,
+  kEpmpEntry12,
+  kEpmpEntry13,
+  kEpmpEntry14,
+  kEpmpEntry15,
+} epmp_entry_t;
+
+/**
+ * EPMP entry count.
+ */
+typedef enum epmp_entry_count {
+  kEpmpEntryCount1,
+  kEpmpEntryCount2,
+  kEpmpEntryCount3,
+  kEpmpEntryCount4,
+  kEpmpEntryCount5,
+  kEpmpEntryCount6,
+  kEpmpEntryCount7,
+  kEpmpEntryCount8,
+  kEpmpEntryCount9,
+  kEpmpEntryCount10,
+  kEpmpEntryCount11,
+  kEpmpEntryCount12,
+  kEpmpEntryCount13,
+  kEpmpEntryCount14,
+  kEpmpEntryCount15,
+  kEpmpEntryCount16,
+} epmp_entry_count_t;
+
+/**
+ * EPMP transaction.
+ *
+ * Must be initialized using epmp_transaction_init before use.
+ *
+ * This structure should be considered opaque.
+ */
+typedef struct epmp_transaction {
+  epmp_entry_count_t remaining;
+  epmp_entry_count_t done;
+  epmp_entry_count_t total;  // remaining + done
+  // TODO: add checksum tracker
+} epmp_transaction_t;
+
+/**
+ * EPMP generic status codes.
+ *
+ * These error codes can be used by any function. However if a function
+ * requires additional status codes, it must define a set of status codes to
+ * be used exclusively by such function.
+ */
+typedef enum epmp_result {
+  kEpmpOk = 0, /**< Success. */
```

Comment:
Throughout: We've been kinda consistent in using 
```
/**
 *
 */
```
comments throughout. (I'd just copy-paste a result enum from one of the DIFs.)

Reply:
Ack. (This is copied and pasted from the original PMP interface. Looks like there are a few other places in the tree this style is used too if you want to change them.)

Reply:
Done